infra/utils/generate-variant-tests.js

```diff
@@ -0,0 +1,18 @@
+/**
+ * This is a helper function that will auto-generate mocha unit tests
+ * for various inputs.
+ *
+ * @param {String} itTitle This is the title that will be passed to the it()
+ * function. The variant will be added to the end of this title to help
+ * idenfity the failing test.
+ * @param {Array<Objects>} variants This should be all the variations of the
+ * test you wish to generate.
+ * @param {function} func This is the function that will be called, with a
+ * variant as the only argument. This function should perform the desired test.
+ */
+export default (itTitle, variants, func) => {
+  variants.forEach((variant) => {
+    it(`${itTitle}. Variant: '${JSON.stringify(variant)}'`,
+      () => func(variant));
```

Comment:
This should be `function() { return func(variant); }` due to the prohibition on using arrow functions for the `it()` test logic.